File: 916-word-subsets/916-word-subsets.java
class Solution {
    public List<String> wordSubsets(String[] words1, String[] words2) {
        List<String> res = new ArrayList<>();
        int[] globalCounter = new int[26];
        for(String word : words2){
            int[] count = new int[26];
            for(int i=0;i<word.length();i++){
                count[word.charAt(i)-'a']++;
            }
            for(int i=0;i<26;i++){
                globalCounter[i] = Math.max(globalCounter[i], count[i]);
            }
        }
        
        for(String word : words1){
            int[] count = new int[26];
            for(int i=0;i<word.length();i++){
                count[word.charAt(i)-'a']++;
            }
            int j=0;
            while(j < 26){
                if(globalCounter[j] > count[j])
                    break;
                j++;
            }
            if(j==26)
                res.add(word);
        }
        return res;
            
    }
}

File: 916-word-subsets/916-word-subsets.py
class Solution:
    def wordSubsets(self, words1: List[str], words2: List[str]) -> List[str]:
        global_counter = {}
        res = []
        for word in words2:
            count = collections.Counter(word)
            for key in count.elements():
                if key not in global_counter:
                    global_counter[key] = 0
                global_counter[key]=max(global_counter[key], count[key])
        for word in words1:
            count = collections.Counter(word)
            flag = True
            for key, val in global_counter.items():
                if key not in count.elements() or count[key] < val:
                    flag = False
                    break
            if flag:
                res.append(word)
        return res
                
                
            
        